pkg/catalog/types.go

```diff
@@ -116,9 +116,16 @@ type MeshCataloger interface {
 	// ListMonitoredNamespaces lists namespaces monitored by the control plane
 	ListMonitoredNamespaces() []string
 
-	// GetPortToProtocolMappingForService returns a mapping of the service's ports to their corresponding application protocol
+	// GetPortToProtocolMappingForService returns a mapping of the service's ports to their corresponding application protocol.
+	// The ports returned are the actual ports on which the application exposes the service derived from the service's endpoints,
+	// ie. 'spec.ports[].targetPort' instead of 'spec.ports[].port' for a Kubernetes service.
 	GetPortToProtocolMappingForService(service.MeshService) (map[uint32]string, error)
```

Comment:
Should we rename this function to provide more clarity ? Given that we have two functions to return ports 
```suggestion
	GetTargetPortToProtocolMappingForService(service.MeshService) (map[uint32]string, error)
```

Reply:
TargetPort is very specific to k8s, while the API is agnostic of providers if you look into it.

Reply:
Based on discussions with @snehachhabria, we will rename the two APIs for more clarity in a subsequent PR.